### DISCUSSION TITLE
端口复用协议测试问题

### DISCUSSION THREAD

**Discussion by 4d4d4d4** [apache/dubbo#14267] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

nacos 2x
dubbo 3x
jdk 8

我在测试端口复用的时候遇到这么一个问题 我在生产者配置文件中

这样配置，期望通过不同协议调用
`dubbo:
  protocol:
    name: dubbo
    port: -1
    ext-protocol: tri
  registry:
    address: nacos://127.0.0.1:8848
  application:
    name: first

`
消费者测试
`    @DubboReference(application = "first", protocol = "dubbo")
    private PortInterface dubboInterface;
    @DubboReference(application = "first", protocol = "tri")
    private PortInterface triInterface;

    @Override
    public void run(String... args) throws Exception {
        System.out.println(triInterface.sayHi("tri"));
        System.out.println(dubboInterface.sayHi("dubbo"));
    }`
结果：只能尝试以生产者中ext-protocol中的协议获取方法会报错,方法仍可以调用（应该是dubbo按协议匹配不到会给我一个）
java.lang.IllegalStateException: urls to invokers error. invokerUrls.size :1, invoker.size :0. urls :[dubbo://192.168.40.1:20880/com.dubbo.port.PortInterface?application=first&background=false&category=providers&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&ext.protocol=tri&file-cache=true&generic=false&interface=com.dubbo.port.PortInterface&ispuserver=true&methods=sayHi&path=com.dubbo.port.PortInterface&pid=14396&prefer.serialization=fastjson2,hessian2&protocol=dubbo&release=3.3.0-beta.2&service-name-mapping=true&side=provider&sticky=false&timestamp=1717154514169&unloadClusterRelated=false]
![Uploading image.png…]()
这样配置，期望通过不同协议调用

消费者测试

结果：只能尝试以生产者中ext-protocol中的协议获取方法会报错,方法仍可以调用（应该是dubbo按协议匹配不到会给我一个）
java.lang.IllegalStateException: urls to invokers error. invokerUrls.size :1, invoker.size :0. urls :[dubbo://192.168.40.1:20880/com.dubbo.port.PortInterface?application=first&background=false&category=providers&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&ext.protocol=tri&file-cache=true&generic=false&interface=com.dubbo.port.PortInterface&ispuserver=true&methods=sayHi&path=com.dubbo.port.PortInterface&pid=14396&prefer.serialization=fastjson2,hessian2&protocol=dubbo&release=3.3.0-beta.2&service-name-mapping=true&side=provider&sticky=false&timestamp=1717154514169&unloadClusterRelated=false]

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
Please submit the discussion in English :)
